src/engine/strat_engine/filesystem.rs

```diff
@@ -47,11 +54,25 @@ impl StratFilesystem {
         }
     }
 
-    pub fn check(&self, dm: &DM) -> EngineResult<FilesystemStatus> {
+    /// check if filesystem is getting full and needs to be extended
+    /// TODO: deal with the thindev in a Fail state.
+    pub fn check(&mut self, dm: &DM) -> EngineResult<FilesystemStatus> {
         match try!(self.thin_dev.status(dm)) {
-            ThinStatus::Good((_mapped, _highest)) => {
-                // TODO: check if filesystem is getting full and might need to
-                // be extended (hint: use statfs(2))
+            ThinStatus::Good((_, _)) => {
```

Comment:
It doesn't matter that it's a tuple if you ignore it entirely, You should be able to just do:

```ThinStatus::Good(_) => ```